### DISCUSSION TITLE
Match monitor by serial number

### DISCUSSION THREAD

**Discussion by dantefromhell** [hyprwm/Hyprland#3115] [General]
I use a Framework laptop as my daily driver - it comes with a USB-C to HDMI card that can be inserted into 4 different slots, resulting in different display names (DP-1 to DP-4) depending on the card slot used.

Since I want a specific setup with my secondary display my thought was matching it by serial number instead.
According to https://github.com/hyprwm/Hyprland/issues/2464 this should be possible, but I can't get it working.

Neither was I able to find sample code. Does anyone got this working and is willing to share their config setting?

**Comment by MightyPlaza** [comment]
desc?

**Reply by dantefromhell** [reply]
> desc?

maybe.... I have multiple of these monitors around my desk - some rotated, so matching serial number is the only way to match a specific display.

The `desc` field contains the serialnumber (good) but also the port (bad), so it depends on how the matching is actually done. The documentation doesnt really state that.
Is it regex matching?

**Reply by MightyPlaza** [reply]
you're supposed to remove the port in the config
![image](https://github.com/hyprwm/Hyprland/assets/123664421/31c2fa4f-ec70-446f-a9cf-345b681b7922)

**Reply by dantefromhell** [reply]
Yeah I know, but the documentation sadly doest not explain how the matching works and which fields are actually supported. Given the issue I linked in the inital post I am not the only person who struggled with this.

The example is not helpful to me, I have to guess and fumble around to actually understand how the output of `hyprctl monitors` was transformed to be used as a rule.

**Reply by MightyPlaza** [reply]
not sure what you mean

**Reply by rowanG077** [reply]
I have the same question. Essentially what you need to know is that in the output of `hyprctl` if you want to match the `description` field you have to use `desc` in your config. What do you need to use in your config to match `serial`.

**Reply by rowanG077** [reply]
This seems to be implemented here: https://github.com/hyprwm/Hyprland/blob/14a3c939ce019623126893aa4c454e62bb29779c/src/Compositor.cpp#L1991

There is no matching rule for anything like serial(or in fact any other field except description). So it doesn't seem possible to match on any other field including serial.

**Reply by dantefromhell** [reply]
@rowanG077 
Yeah I came to the same conclusion.

For my case it worked because the monitor serial number is mentioned in the `description` field, so I the following line correctly matches my specific monitor
```
monitor=desc:<DESCRIPTION_STRING>,1920x1080,0x0,1
```
with `DESCRIPTION_STRING` copied from `hyprctl monitors` `description` field output.

**Reply by ElianHernandez19** [reply]
chad
